src/benchq/problem_ingestion/molecule_instance_generation.py

```diff
@@ -48,49 +49,111 @@ class ChemistryApplicationInstance:
+                and whose second is the meanfield object containing the SCF solution.
+        """
+        molecule = self.get_pyscf_molecule()
+        mean_field_object = (scf.RHF if self.multiplicity == 1 else scf.ROHF)(molecule)
+        mean_field_object.run()
+        if self.avas_atomic_orbitals or self.avas_minao:
+            molecule, mean_field_object = truncate_with_avas(
+                mean_field_object,
+                self.avas_atomic_orbitals,
+                self.avas_minao,
             )
-        )
+        return molecule, mean_field_object
 
     def get_active_space_hamiltonian(self) -> openfermion.InteractionOperator:
         """Generate the fermionic Hamiltonian corresponding to the instance's
-        active space."""
-        if self.avas_atomic_orbitals or self.avas_minao:
-            raise ValueError(
-                "Generating the active space Hamiltonian for application instances"
-                "with AVAS is not currently supported."
-            )
-        return self.get_molecular_data().get_molecular_hamiltonian(
+        active space.
+
+        Returns:
+            The fermionic Hamiltonian corresponding to the instance's active space. Note
+                that the active space will account for both AVAS and the
+                occupied_indices/active_indices attributes.
+        """
+        return self._get_molecular_data().get_molecular_hamiltonian(
             occupied_indices=self.occupied_indices,
             active_indices=self.active_indices,
         )
 
     def get_active_space_meanfield_object(self) -> scf.hf.SCF:
         """Run an SCF calculation using PySCF and return the results as a meanfield
-        object."""
+        object.
+
+        Currently, this method does not support the occupied_indices and active_indices
+        attributes and will raise an exception if they are set.
+
+        Returns:
+            A meanfield object corresponding to the instance's active space, accounting
+                for AVAS."""
         if self.active_indices or self.occupied_indices:
             raise ValueError(
                 "Generating the meanfield object for application instances with "
                 "active and occupied indices is not currently supported."
             )
-        if not self.avas_atomic_orbitals or not self.avas_minao:
-            raise ValueError(
-                "Generating the meanfield object for application instances without "
-                "avas_atomic_orbitals and avas_minao is not currently supported."
-            )
-        return truncate_with_avas(
-            self.get_molecular_data()._pyscf_data["scf"],
-            self.avas_atomic_orbitals,
-            self.avas_minao,
+        return self._run_pyscf()[1]
+
+    def _get_molecular_data(self):
+        """Given a PySCF meanfield object and molecule, return a PyscfMolecularData object.
+
+        Args:
```

Comment:
I think the list of args is redundant here.

Reply:
Good catch, fixed in a2216750a1c9042466a4c39aff99db5299e1afe9